app/src/main/java/com/duckduckgo/app/httpsupgrade/HttpsUpgrader.kt

```diff
@@ -54,22 +55,33 @@ class HttpsUpgraderImpl(
             return false
         }
 
-        if (whitelistedDao.contains(uri.host)) {
-            Timber.d("${uri.host} is in whitelist and so not upgradable")
+        val host = uri.host
+        if (whitelistedDao.contains(host)) {
+            Timber.d("${host} is in whitelist and so not upgradable")
             return false
         }
 
         httpsBloomFilter?.let {
-            val shouldUpgrade = it.contains(uri.host)
-            Timber.d("${uri.host} ${if (shouldUpgrade) "is" else "is not"} upgradable")
+
+            val initialTime = System.nanoTime()
+            val shouldUpgrade = it.contains(host)
+            val totalTime = System.nanoTime() - initialTime
+            Timber.d("${host} ${if (shouldUpgrade) "is" else "is not"} upgradable, lookup in ${totalTime/NANO_TO_MILLIS_DIVISOR}ms")
```

Comment:
Is it nicer to use `TimeUnit` here for the conversion rather than doing it manually?

`TimeUnit.NANOSECONDS.toMillis(totalTime)` ?

Reply:
Sadly not. TimeUnit conversions only return whole numbers, so would always spit out zero.